doc/tutorial/single_cell_recipe.rst

```diff
@@ -0,0 +1,181 @@
+
+    # (3) Create cell and set properties
+    decor = arbor.decor()
+    decor.set_property(Vm=-40)
+    decor.paint('"soma"', 'hh')
+    decor.place('"center"', arbor.iclamp( 10, 2, 0.8))
+    decor.place('"center"', arbor.spike_detector(-10))
+    cell = arbor.cable_cell(tree, labels, decor)
+
+The recipe
+----------
+
+The :class:`arbor.single_cell_model` of the previous example created a :class:`arbor.recipe` under
+the hood, and abstracted away the details so we were unaware of its existence.
+
+Creating an analogous recipe starts with creating a class that inherits from :class:`arbor.recipe` and overrides and implements some of :class:`arbor.recipe` methods. Not all methods have to be overridden, but some will always have to be, such as :meth:`arbor.recipe.num_cells`. It returns `0` by default and models without cells are quite boring!
+
+.. code-block:: python
+
+    # (4) Define a recipe for a single cell and set of probes upon it.
+    # This constitutes the corresponding generic recipe version of
+    # `single_cell_model.py`.
+
+    class single_recipe (arbor.recipe):
+        def __init__(self, cell, probes):
+            # (4.1) The base C++ class constructor must be called first, to ensure that
+            # all memory in the C++ class is initialized correctly.
+            arbor.recipe.__init__(self)
+            self.the_cell = cell
+            self.the_probes = probes
+            self.the_props = arbor.neuron_cable_propetries()
+            self.the_cat = arbor.default_catalogue()
+            self.the_props.register(self.the_cat)
+
+        def num_cells(self):
+            # (4.2) Override the num_cells method
+            return 1
+
+        def num_sources(self, gid):
+            # (4.3) Override the num_sources method
+            return 1
+
+        def cell_kind(self, gid):
+            # (4.4) Override the cell_kind method
+            return arbor.cell_kind.cable
+
+        def cell_description(self, gid):
+            # (4.5) Override the cell_description method
+            return self.the_cell
+
+        def probes(self, gid):
+            # (4.6) Override the probes method
+            return self.the_probes
+
+        def global_properties(self, kind):
+            # (4.7) Override the global_properties method
+            return self.the_props
+
+Step **(4)** describes the recipe that will reflect our single cell model.
+
+Step **(4.1)** defines the class constructor. It can take any shape you need, but it is important to call base class' constructor. In general it's a good idea to make objects that'll be return by methods that need to be overridden members of the class. With this constructor, we could easily change the cell and probes of the model, should be want to do so. Here we initialize the cell properties to match Neuron's defaults using Arbor's built-in :meth:`arbor.neuron_cable_properties` and extend with Arbor's own :meth:`arbor.default_catalogue`.
```

Comment:
Can we split the sentence over several lines? 
Also: 
- `that'll be return*ed*`
- Whole sentence is ambiguous: `In general it's a good idea to make objects that'll be return by methods that need to be overridden members of the class` -> ` If the overridden methods of the class need to return an object, it may be a good idea to have the returned object be a member of the class`. 
- `should *we* want to do so`